src/crawlee/crawlers/_basic/_basic_crawler.py

```diff
@@ -1256,52 +1278,51 @@ def _convert_url_to_request_iterator(self, urls: Sequence[str | Request], base_u
             else:
                 yield Request.from_url(url)
 
+    def _get_context_aware_requests(
```

Comment:
day 4237 of me hating methods that start with `get_` 😁 the method does not really retrieve anything without modifying it, the logic is much more sophisticated. I'd either rename it to signal that the `context` is used to handle relative URLs, or just inline it in the `_add_requests` body with a comment where you have more creative freedom than in a method name